caffe2/core/dispatch/DeviceId.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <functional>
+#include <iostream>
+
+namespace c10 {
+
+enum class DeviceId : uint8_t {
```

Comment:
Let's call it DeviceType or something like that. DeviceId can be misinterpreted  as GPU's index